apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/AppController.java

```diff
@@ -47,6 +51,30 @@ public AppController(
     this.appOpenApiService = appOpenApiService;
   }
 
+  /**
+   * @see com.ctrip.framework.apollo.portal.controller.AppController#create(AppModel)
+   */
+  @PreAuthorize(value = "@consumerPermissionValidator.hasCreateApplicationPermission(#request)")
+  @PostMapping(value = "/apps")
+  public void createApp(
+      @RequestBody OpenCreateAppDTO req,
+      HttpServletRequest request
+  ) {
+    if (null == req.getAppId()) {
+      throw new BadRequestException("AppId is null");
+    }
+    if (req.isAssignAppRoleToSelf()) {
+      // create app and assign app role to this consumer
+      this.appOpenApiService.createApp(req);
+      // todo @Transactional
+      long consumerId = this.consumerAuthUtil.retrieveConsumerId(request);
+      consumerService.assignAppRoleToConsumer(consumerId, req.getAppId());
```

Comment:
`consumerId` cannot pass to `appOpenApiService`, should we add it to `appOpenApiService`'s `createApp` method signture? But user don't know what is `consumerId`, user only know about `token`.

And the database Transactional break here, can we use ThreadLocal or another way to pass `consumerId`, and delete it after consumerService.assignAppRoleToConsumer invoked.

Reply:
We may mark the `@Transactional` to this method?

```java
  @Transactional
  @PreAuthorize(value = "@consumerPermissionValidator.hasCreateApplicationPermission(#request)")
  @PostMapping(value = "/apps")
  public void createApp(
      @RequestBody OpenCreateAppDTO req,
      HttpServletRequest request
  ) {
```

Reply:
Done.